shared/utility/math/filter/UKF.hpp

```diff
@@ -177,27 +187,33 @@ namespace utility::math::filter {
             covariance_weights.fill(1.0 / (2.0 * covariance_sigma_weight));
             covariance_weights[0] = lambda / covariance_sigma_weight + (1.0 - (alpha * alpha) + beta);
 
-            set_state(initial_mean, initial_covariance);
+            return set_state(initial_mean, initial_covariance);
         }
 
-        void set_state(StateVec initial_mean, StateMat initial_covariance) {
+        [[nodiscard]] bool set_state(StateVec initial_mean, StateMat initial_covariance) {
             mean       = initial_mean;
             covariance = initial_covariance;
 
             // Calculate our sigma points
-            sigma_mean   = mean;
-            sigma_points = generate_sigma_points(mean, covariance, covariance_sigma_weight);
+            sigma_mean                  = mean;
+            auto&& points_chole_success = generate_sigma_points(mean, covariance, covariance_sigma_weight);
+
+            sigma_points = points_chole_success.first;
 
             // Reset our state for more measurements
             covariance_update = covariance_weights.asDiagonal();
             d.setZero();
             centred_sigma_points = sigma_points.colwise() - sigma_mean;
+            return points_chole_success.second;
         }
 
         template <typename... Args>
-        void time(const Scalar& dt, const Args&... params) {
+        [[no_discard]] bool time(const Scalar& dt, const Args&... params) {
             // Generate our sigma points
-            sigma_points = generate_sigma_points(mean, covariance, covariance_sigma_weight);
+            auto&& points_chole_success = generate_sigma_points(mean, covariance, covariance_sigma_weight);
```

Comment:
`points_chole_success` makes be barf every time I read it :vomiting_face:

Reply:
I find naming a pair hard as hahaha. do you have any suggestions?